core/src/main/scala-3/caliban/schema/SchemaDerivation.scala

```diff
@@ -23,6 +23,14 @@ object PrintDerived {
 trait CommonSchemaDerivation {
   export DerivationUtils.customizeInputTypeName
 
+  /**
+   * Enables the `SemanticNonNull` feature on derivation.
+   * It is currently disabled by default since it is not yet stable.
+   *
+   * Override this method and return `true` to enable the feature.
+   */
+  def enableSemanticNonNull: Boolean = false
```

Comment:
May I recommend we use a `DerivationConfig` case class instead? I can see us wanting to add more derivation customization options in the future

Reply:
https://github.com/ghostdogpr/caliban/pull/2180/commits/d8d3da75ee71fc465c3f49979927a5b5eeefebcb